processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -200,6 +200,9 @@ public File persist(
                 indexSpec.getBitmapSerdeFactory().getBitmapFactory()
             )
         ),
+        // if index is not rollup, then it should be not rollup here
+        // if index is rollup, then it is no need to rollup again
+        false,
```

Comment:
In persist stage, if index.rollup==true, persist.rollup==false, then the rowMergerFn(the MergeIterable) actually doesn't change anything(element self and the order) due to having only one underlying iterator, in this case, everything works the same as before.
In segments merge stage, merge.rollup==true, everything should be the same as before due to it is the fallback case before this patch